index-refactor.py

```diff
@@ -0,0 +1,143 @@
+Rekease_url = "https://api.github.com/repos/actions/runner/releases/latest"
+#yaml File path
+file_path = "runner-test.txt"
+#sender email for the mail notification.
+#sender_email = "geetha.palyam@gmail.com"
+# Gmail app password
+# password = 'kndtknppilckttof'
+
+client = boto3.client('ssm')  # Creates an Amazon Simple Systems Manager client
+sender_email = client.get_parameter(Name='/test/check/sender_email', WithDecryption=True)["Parameter"]["Value"]
+password = client.get_parameter(Name='/test/check/sender_email_password', WithDecryption=True)["Parameter"]["Value"]
+Access_token = client.get_parameter(Name='/test/check/access_token', WithDecryption=True)["Parameter"]["Value"]
+webhook_url = client.get_parameter(Name='/test/check/webhook', WithDecryption=True)["Parameter"]["Value"]
+
+print("sender_email", sender_email)
+print("password", password)
+# Email notification reciever
+receiver_email = "palyamgeetha@gmail.com"
+#branch name
+branch_name = "runner"
+#slack channel receives notification
+channel_name = "#git-script"
+
+def sendSlackNotification(pr_number):
+    # Code to send alert to the slack channel
+    url = webhook_url
+    message = ("new Pull Request Link: https://github.com/{}/{}/pull/{}".format(username, repoName, pr_number))
+    title = (f"New Incoming pull Request :zap:")
+    slack_data = {
+        "username": "NotificationBot",
+        "icon_emoji": ":satellite:",
+        "channel" : channel_name,
+        "attachments": [
+            {
+                "color": "#9733EE",
+                "fields": [
+                    {
+                        "title": title,
+                        "value": message,
+                        "short": "false",
+                    }
+                ]
+            }
+        ]
+    }
+    byte_length = str(sys.getsizeof(slack_data))
+    headers = {'Content-Type': "application/json", 'Content-Length': byte_length}
+    response = requests.post(url, data=json.dumps(slack_data), headers=headers)
+    if response.status_code != 200:
+        raise Exception(response.status_code, response.text)
+
+def sendEmailNotification(pr_number):
+    # Code to send alert via email
+    smtp_server = "smtp.gmail.com"
+    port = 587  # For starttls
+
+    url = "https://github.com/{}/{}/pull/{}".format(username, repoName, pr_number)
+    message = '''hello
+
+    A new Pull Request has been created
+
+    The URL is: '''+url
+
+    # Create a secure SSL context
+    context = ssl.create_default_context()
+
+    # Try to log in to server and send email
+    try:
+        server = smtplib.SMTP(smtp_server,port)
+        server.ehlo() # Can be omitted
+        server.starttls(context=context) # Secure the connection
+        server.ehlo() # Can be omitted
+        server.login(sender_email, password)
+        server.sendmail(sender_email, receiver_email, message)
+    except Exception as e:
+        # Print any error messages to stdout
+        print(e)
+    finally:
+        server.quit()
```

Comment:
can be refactored to the following 

    smtp = smtplib.SMTP_SSL(host="smtp.sendgrid.net", port=465)
    sendgrid_user, sendgrid_pw  = get_email_credentials()
    smtp.login(user=sendgrid_user, password=sendgrid_pw)
    smtp.send_message(msg, msg['From'], email)
    smtp.quit()


also, do we need to keep  server.ehlo() # Can be omitted?

Reply:
ok